pkg/kudoctl/packages/install/install.go

```diff
@@ -0,0 +1,144 @@
+// Package install provides function to install package resources
+// on a Kubernetes cluster.
+package install
+
+import (
+	"strings"
+	"time"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/clog"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/packages"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/kudo"
+)
+
+type Options struct {
```

Comment:
I'd like to adopt the _type.go-pattern_ as with other packages and keep all general types in an `install/types.go`. It keeps entangling cyclic dependencies later much easier.

Reply:
Adding a `install.types.go` won't change anything regarding cyclic dependencies, because it would still be in the same package. For this it would need to be in `types.go` of the `packages` package. I consider this an anti-pattern, types should be declared close to the functions using this type. Cyclic dependencies can be resolved by good package management following SRP.

Reply:
> Adding a install.types.go won't change anything regarding cyclic dependencies
not yet, that's why I wrote that it becomes easier _"later"_  😉 

> I consider this an anti-pattern
well, k8s codebase disagrees with you, and frankly so do I. we adopted this pattern for a good reason. Lack of fine-grained imports in Go makes it challenging to avoid cyclic dependencies.

Reply:
I have yet to see where the Kubernetes code base disagrees. Their usage of `types.go` is done to provide consistency when using generators like `deepcopy-gen` or other kubebuilder tools on structures that are supposed to be used as part of the Kubernetes API. Which isn't the case here.

Reply:
The can be multiple reasons for that. And again, being forced to resolve multiple cyclic dependencies (and giving up on at least one of them) in our codebase makes me appreciate this pattern a lot. Especially in a package like `install` which is potentially used by server and client and itself uses server types (like Instance). But I guess we'll have to agree to disagree here.